code/modules/medical_machinery/machines/advanced_medical_machinery.dm

```diff
@@ -0,0 +1,92 @@
+/obj/machinery/medical/dialysis
+	name = "Direct Blood Filtration Unit"
+	desc = "Automatically filtrates your blood from all chemicals."
+	icon_state = "dialysis"
+	///Amount of purged chems per process
+	var/purge_amount = 3
+
+/obj/machinery/medical/dialysis/RefreshParts()
+	var/change = 0
+	for(var/obj/item/stock_parts/capacitor/C in component_parts)
+		change += C.rating
+	purge_amount = initial(purge_amount) * change
+	return
```

Comment:
you probably want to carefully consider how to balance this

component ratings range from 1-4, meaning our base purge PER TICK is 3-12 units of EACH REAGENT

that seems excessive

Reply:
yeah, i agree ill either nerf it , or make it use more energy the better parts it has